services/movie_session.py

```diff
@@ -0,0 +1,59 @@
+from datetime import datetime
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession
+
+
+def create_movie_session(
+    movie_show_time: datetime,
+    movie_id: int,
+    cinema_hall_id: int
+) -> None:
+    MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id
+    )
+
+
+def get_movies_sessions(
+    session_date: str | None = None,
+) -> QuerySet:
+    queryset = MovieSession.objects.all()
+
+    if session_date:
+        queryset = queryset.filter(
+            show_time__date=datetime.strptime(
+                session_date,
+                "%Y-%m-%d"
+            ).date()
+        )
+
+    return queryset
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+    session_id: int,
+    show_time: datetime | None = None,
+    movie_id: int | None = None,
+    cinema_hall_id: int | None = None,
+) -> None:
+    queryset = MovieSession.objects.all().filter(id=session_id)
```

Comment:
In the `update_movie_session` function, using `.filter()` is not ideal for updating a single instance. Consider using `.get()` to retrieve the instance and then update its fields directly to ensure you're working with a single instance.